### PR TITLE
Default select of report

### PR DESCRIPTION
**Pull request by shreya-fyle1**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "@capacitor/app": "^1.1.1",
     "@capacitor/browser": "^1.0.7",
     "@capacitor/camera": "^1.3.1",
-    "@capacitor/core": "^3.4.3",
+    "@capacitor/core": "^3.9.0",
```
Comment (Dimple16) [inline package.json:46]: you can remove package.json files from the PR

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { Value } from './../../core/models/v2/stats-response.model';
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1]: why is this added here?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -224,6 +227,8 @@ export class AddEditExpensePage implements OnInit {
 
   matchingCCCTransactions = [];
 
+  listofreports = [];
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:230]: remove this as you aren't using it

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -120,6 +121,8 @@ export class AddEditExpensePage implements OnInit {
 
   @ViewChild('fileUpload', { static: false }) fileUpload: any;
 
+  displayValue;
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:124]: which change is this related to?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 5]
```diff
@@ -349,6 +354,8 @@ export class AddEditExpensePage implements OnInit {
 
   breakfastSystemCategories: string[];
 
+  autoSubmissionReportEnabled;
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:357]: add the corresponding model type here

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 6]
```diff
@@ -2816,7 +2836,9 @@ export class AddEditExpensePage implements OnInit {
     });
 
     this.autoSubmissionReportName$ = this.reportService.getAutoSubmissionReportName();
-
+    this.autoSubmissionReportName$.subscribe((res) => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2839]: res -> `autoSubmissionReportEnabled`

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

**Review by Dimple16 — CHANGES_REQUESTED** [review]
remove package files from the PR

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 7]
```diff
@@ -2816,7 +2831,9 @@ export class AddEditExpensePage implements OnInit {
     });
 
     this.autoSubmissionReportName$ = this.reportService.getAutoSubmissionReportName();
-
+    this.autoSubmissionReportName$.subscribe((autoSubmissionReportEnabled) => {
+      this.autoSubmissionReportEnabled = autoSubmissionReportEnabled;
```
[the remark this answers is not in the document]
Comment (yksolanki9) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2835]: A new variable is not needed, you can directly use forkJoin as mentioned above

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2835]: remove this block as it isn't used anymore

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 8]
```diff
@@ -349,6 +349,8 @@ export class AddEditExpensePage implements OnInit {
 
   breakfastSystemCategories: string[];
 
+  autoSubmissionReportEnabled: string;
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:352]: Remove unused variable

**Review by yksolanki9 — APPROVED** [review]
Minor comment

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 9]
```diff
@@ -1396,6 +1401,16 @@ export class AddEditExpensePage implements OnInit {
                 .find((reportOption) => reportOption.rp.id === this.activatedRoute.snapshot.params.rp_id)
             )
           );
+        } else if (!autoSubmissionReportName) {
+          return this.reports$.pipe(
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:1405]: this isn't resolved yet

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 8 above)
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:352]: remove this if unused

**Review by Dimple16 — CHANGES_REQUESTED** [review]

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**